src/pal/src/configure.cmake

```diff
@@ -366,6 +366,9 @@ check_cxx_source_runs("
 #include <stdlib.h>
 #include <time.h>
 #include <sys/time.h>
+#ifdef (__linux__)
```

Comment:
The __LINUX__ symbol defined in /CMakeList.txt is not valid in configure.cmake's check_cxx_source_compiles, but the Linux standard __linux__ works in that context of check_cxx_source_compiles.

The preprocessor macro definitions declared by CMakeList.txt is (don't know why, but that's the observation in Linux systems) not effective in check_cxx_source_compiles() while the Linux compiler standard __linux__ works in that context.

Because __linux__ is already defined by compilers, it is correct to use __LINUX__ in CMakeList, but for sources in check_cxx_source_compiles, which does not automatically inherit CMakeList's add_definitions, __linux__ is correct UNLESS we manually add CMAKE_REQUIRED_DEFINITIONS() with another if statements in CMAKE.